canonicalwebteam/blog/django/views.py

```diff
@@ -86,6 +90,68 @@ def group(request, slug, template_path):
     return render(request, template_path, context)
 
 
+def archives(request, template_path="blog/archives.html"):
+    try:
+        page_param = request.GET.get("page", default="1")
+        category_param = request.GET.get("category", default="")
+        group_param = request.GET.get("group", default="")
+        month_param = request.GET.get("month", default="")
+        year_param = request.GET.get("year", default="")
+
+        group = None
+        group_id = ""
+        if group_param != "":
+            group = api.get_group_by_slug(group_param)
+            group_id = group["id"]
+
+        category_id = ""
+        if category_param != "":
+            category = api.get_category_by_slug(category_param)
+            category_id = category["id"]
```

Comment:
I would declare groups and categories = [] beforehand and then append the id to it inside the if.